Confidence/src/main/java/com/spotify/confidence/Confidence.kt

```diff
@@ -77,10 +80,23 @@ class Confidence internal constructor(
         return flagResolver.resolve(flags, getContext())
     }
 
-    suspend fun awaitReconciliation() {
-        if (currentFetchJob != null) {
-            currentFetchJob?.join()
-            activate()
+    suspend fun awaitReconciliation(timeoutMillis: Long = 5000) {
+        try {
+            withTimeout(timeoutMillis) {
+                while (currentFetchJob == null ||
+                    currentFetchJob?.isCompleted == true ||
+                    currentFetchJob?.isCancelled == true
+                ) {
+                    // If this function is called just after a putContext it will likely just delay once.
+                    delay(1)
```

Comment:
I think the un-predictability of when `putContext` is called + the short timespan of `1ms` _could_ make this code quite inefficient

Reply:
Please explain more, not sure I understand the concern.

Reply:
I think what I am getting at is: are there more idiomatic ways to achieve this behaviour? I am looking at `job.invokeOnCompletion`, for example, but I don't know if it does exactly what we need